src/audio/dai-zephyr.c

```diff
@@ -719,145 +697,81 @@ static int dai_capture_params(struct comp_dev *dev, uint32_t period_bytes,
 		prev = dma_block_cfg;
 		prev->next_block = ++dma_block_cfg;
 	}
+
 	if (prev)
 		prev->next_block = dma_cfg->head_block;
-	dd->z_config = dma_cfg;
 
-free:
-	if (err < 0)
-		dma_sg_free(&config->elem_array);
-out:
-	buffer_release(dma_buf);
+	dd->z_config = dma_cfg;
 
-	return err;
+	return 0;
 }
 
 static int dai_params(struct comp_dev *dev, struct sof_ipc_stream_params *params)
 {
-	struct sof_ipc_stream_params hw_params = *params;
 	struct dai_data *dd = comp_get_drvdata(dev);
-	struct comp_buffer __sparse_cache *buffer_c;
-	uint32_t frame_size;
-	uint32_t period_count;
-	uint32_t period_bytes;
-	uint32_t buffer_size;
-	uint32_t addr_align;
-	uint32_t align;
-	int err;
+	struct dma_sg_config *config = &dd->config;
+	uint32_t period_bytes = 0;
+	uint32_t period_count = 0;
+	int err = 0;
 
 	comp_dbg(dev, "dai_params()");
 
-	/* configure dai_data first */
-	err = ipc_dai_data_config(dev);
```

Comment:
isn't this a change in behaviour - at least a theoretic one? Before this patch this function would always be called and would set various persistent configuration fields, after this patch it's only called if the state of the DAI device is correct?